bin/dialog_wrapper.py

```diff
@@ -4,13 +4,28 @@
 import sys
 import dialog
 import traceback
-from StringIO import StringIO
+from io import StringIO
+from os import environ
+from pathlib import Path
 
 email_re = re.compile(r"(?:^|\s).*\S@\S+(?:\s|$)", re.IGNORECASE)
+logfile = Path('/var/log/dialog.log')
+
+
+def _dia_log(msg_str):
+    if 'DIALOG_DEBUG' not in environ.keys():
+        return
+    log_content = ''
+    if logfile.is_file():
+        log_content = logfile.read_text()
+    log_content = log_content + msg_str
+    logfile.write_text(log_content)
```

Comment:
Rather than reading the file, concatening text to the file's content then re-writing the file, we can just append directly by opening the file in `append` mode.

With either `open(path, 'a')` or `Path.open('a')` then just writing `msg_str` directly to that file object.

Unfortunately it doesn't look like pathlib directly supports appending without using open.

Reply:
Also we should probably be adding a newline to the end of msg_str. Both because it makes it simpler to write lines of text (without manually adding `"\n"` to strings used in log calls) And because text files on linux generally should end in a newline.